### PR TITLE
new visualization: "artificial viscosity composition"

### PR DESCRIPTION
**Pull request by tjhei**
This is ripped out of the melt branch. Work was done by @jdannberg, mistakes are my fault of course.

### CODE REVIEW

**Comment by bangerth** [comment]
The new test fails:
```
11/311 Test  #17: artificial_viscosity ...........................................***Failed    0.82 sec
*** "Test artificial_viscosity failed": ***
[1/14] Generating artificial_viscosity.x.prm
[2/14] Generating output-artificial_viscosity/solution-00000.0000.gnuplot.cmp.notime
[3/14] Generating output-artificial_viscosity/solution-00001.0000.gnuplot.cmp.notime
[4/14] Generating output-artificial_viscosity/screen-output.cmp.notime
[5/14] Generating output-artificial_viscosity/statistics.cmp.notime
[6/14] Generating output-artificial_viscosity/screen-output
FAILED: cd /aspect-tester/build-gcc/tests && for i in /aspect-tester/build-gcc/tests/output-artificial_viscosity/*  do if echo  | grep -q -v .cmp.notime  then rm -f   fi  done && /aspect-tester/build-gcc/aspect /aspect-tester/build-gcc/tests/artificial_viscosity.x.prm > /aspect-tester/build-gcc/tests/output-artificial_viscosity/screen-output.tmp && /aspect-tester/aspect/tests/cmake/default.sh screen-output </aspect-tester/build-gcc/tests/output-artificial_viscosity/screen-output.tmp >/aspect-tester/build-gcc/tests/output-artificial_viscosity/screen-output && /usr/bin/perl -pi /aspect-tester/aspect/tests/normalize.pl /aspect-tester/build-gcc/tests/output-artificial_viscosity/*
Line <104> of file <input string>:
    The entry value
        melt global
    for the entry named
        Model name
    does not match the given pattern
        [Selection Morency and Doin|Steinberger|averaging|composition reaction|depth dependent|diffusion dislocation|drucker prager|latent heat|latent heat melt|multicomponent|simple|simple compressible|simpler|unspecified ]
Line <105> of file <input string>: There is no such subsection to be entered: Material model.Melt global
    subsection Material model
      subsection Melt global
Line <106> of file <input string>: No such entry was declared:
    Exponential melt weakening factor
    <Present subsection:
        subsection Material model
    >
Line <107> of file <input string>: No such entry was declared:
    Thermal conductivity
    <Present subsection:
        subsection Material model
    >
...
```

File: include/aspect/postprocess/visualization/artificial_viscosity_composition.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      /**
+       * A class derived that implements a function that provides the
+       * artificial viscosity for the temperature equation on each cell for
```
Comment (bangerth) [inline include/aspect/postprocess/visualization/artificial_viscosity_composition.h]: temperature -> composition

File: source/postprocess/visualization/artificial_viscosity_composition.cc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        {
+          prm.enter_subsection("Visualization");
+          {
+            prm.enter_subsection("Artificial viscosity composition");
+            {
+              const std::string field_name = prm.get("Name of compositional field");
+
+              AssertThrow(this->introspection().compositional_name_exists(field_name),
+                          ExcMessage("The compositional field whose artificial viscosity you want to "
+                                     "visualize does not exist."));
```
Comment (bangerth) [inline source/postprocess/visualization/artificial_viscosity_composition.cc]: I forgot how we do this elsewhere -- do we *require* the symbolic name of a compositional field, or do we *also* allow specifying a number?

Comment (jdannberg) [inline source/postprocess/visualization/artificial_viscosity_composition.cc]: I am not sure if there are other places where we do something like this. Normally we do everything for all of the fields. In principle we could also allow a number.

Comment (tjhei) [inline source/postprocess/visualization/artificial_viscosity_composition.cc]: The only other place I found is visualization "Compositional fields as vectors" and that also requires names. I can add the option to use and index instead if you want (1 based?).

Comment (bangerth) [inline source/postprocess/visualization/artificial_viscosity_composition.cc]: We used to use only numbers for compositional fields until someone (I don't recall who) implemented the translation to names. I think we used 0-based indices before.

But if there are no other places where we do this, let's keep it as you do now. Maybe just rewrite the error message to say "No compositional field with name <...> exists for which you want to visualize the artificial viscosity."

**Comment by tjhei** [comment]
updated.

**Comment by tjhei** [comment]
rebased and good to go.

**Comment by gassmoeller** [comment]
You will need to do another rebase because of #854.

**Pull request merged**